### PR TITLE
remove post validation of rbd device

### PR DESCRIPTION
**Pull request by Madhu-1**
once we map the rbd image on a node we will get the device name in the map output itself, no need to check the device path post rbd  mapping

Signed-off-by: Madhu Rajanna <madhupr007@gmail.com>

### CODE REVIEW

**Comment by Madhu-1** [comment]
testing pending for nbd

**Comment by humblec** [comment]
> testing pending for nbd

LGTM. However will wait for `NBD` testing too.

**Comment by Madhu-1** [comment]
works for  nbd-rbd
```
sh-4.2# rbd-nbd map replicapool/test                   
/dev/nbd1
```

File: pkg/rbd/rbd_attach.go [hunk 1]
```diff
@@ -282,16 +280,7 @@ func createPath(volOpt *rbdVolume, cr *util.Credentials) (string, error) {
...
-	if !found {
-		output, err := execCommand(cmdName, []string{
-			"unmap", imagePath, "--id", cr.ID, "-m", volOpt.Monitors, "--keyfile=" + cr.KeyFile})
-		if err != nil {
-			klog.Warningf("rbd: unmap error %v, rbd output: %s", err, string(output))
-		}
-		return "", fmt.Errorf("could not map image %s, Timeout after 10s", imagePath)
-	}
-	return devicePath, nil
+	return string(output), nil
```
Comment (ShyamsundarR) [inline pkg/rbd/rbd_attach.go]: output will contain an extra newline at the end of the string, is this acceptable, or should we trim it?

Comment (Madhu-1) [reply pkg/rbd/rbd_attach.go]: if it contains extra newline the CI should have failed isn't it?

Comment (ShyamsundarR) [reply pkg/rbd/rbd_attach.go]: FWIW, I tested the output, before posting the comment, and it does contain an extra newline. The question is should we trim it or not. CI may not fail, but do we want the new line byte in the string that we are using, and hence may fail other comparisons down the line or is it better to be safe and store/return what we want without the new line?

Comment (Madhu-1) [reply pkg/rbd/rbd_attach.go]: in that case, we can trim it, updated PR

Comment (ShyamsundarR) [reply pkg/rbd/rbd_attach.go]: Thank you!

**Comment by ShyamsundarR** [comment]
CI failure is as below and looks unrelated (I saw one other instance of the same in @Daniel-Pivonka 's PR, so may need some attention in time). For now retriggering the job, as I want to rebase my PR on top of this.

Error:
```
Aug  7 14:05:47.957: INFO: failed to delete resource in  with err error running &{/usr/local/bin/kubectl [kubectl --kubeconfig=/home/travis/.kube/config delete -f ../deploy/rbd/kubernetes/v1.14+/helm] []  <nil>  unable to decode "../deploy/rbd/kubernetes/v1.14+/helm/Chart.yaml": Object 'Kind' is missing in '{"apiVersion":"v1","appVersion":"1.0.0","description":"Container Storage Interface (CSI) driver, provisioner, snapshotter, and attacher for Ceph RBD","home":"https://github.com/ceph/ceph-csi","keywords":["ceph","rbd","ceph-csi"],"name":"ceph-csi-rbd","sources":["https://github.com/ceph/ceph-csi/tree/csi-v1.0/deploy/rbd/helm"],"version":"0.8.0"}'

unable to decode "../deploy/rbd/kubernetes/v1.14+/helm/values.yaml": Object 'Kind' is missing in '{"attacher":{"affinity":{},"enabled":true,"image":{"pullPolicy":"IfNotPresent","repository":"quay.io/k8scsi/csi-attacher","tag":"v1.2.0"},"name":"attacher","nodeSelector":{},"resources":{},"tolerations":[]},"configMapName":"ceph-csi-config","driverName":"rbd.csi.ceph.com","nodeplugin":{"affinity":{},"name":"nodeplugin","nodeSelector":{},"plugin":{"image":{"pullPolicy":"IfNotPresent","repository":"quay.io/cephcsi/cephcsi","tag":"canary"},"resources":{}},"registrar":{"image":{"pullPolicy":"IfNotPresent","repository":"quay.io/k8scsi/csi-node-driver-registrar","tag":"v1.1.0"},"resources":{}},"tolerations":[]},"pluginDir":"/var/lib/kubelet/plugins","provisioner":{"affinity":{},"image":{"pullPolicy":"IfNotPresent","repository":"quay.io/k8scsi/csi-provisioner","tag":"v1.3.0"},"name":"provisioner","nodeSelector":{},"replicaCount":3,"resources":{},"tolerations":[]},"rbac":{"create":true},"registrationDir":"/var/lib/kubelet/plugins_registry","serviceAccounts":{"attacher":{"create":true,"name":null},"nodeplugin":{"create":true,"name":null},"provisioner":{"create":true,"name":null}},"snapshotter":{"image":{"pullPolicy":"IfNotPresent","repository":"quay.io/k8scsi/csi-snapshotter","tag":"v1.2.0"},"resources":{}},"socketDir":"/var/lib/kubelet/plugins/rbd.csi.ceph.com","socketFile":"csi.sock"}'

 [] <nil> 0xc0009abda0 exit status 1 <nil> <nil> true [0xc000972598 0xc0009725c8 0xc000972608] [0xc000972598 0xc0009725c8 0xc000972608] [0xc0009725b8 0xc0009725f8] [0x7ba2e0 0x7ba2e0] 0xc000138780 <nil>}:

Command stdout:

stderr:

unable to decode "../deploy/rbd/kubernetes/v1.14+/helm/Chart.yaml": Object 'Kind' is missing in '{"apiVersion":"v1","appVersion":"1.0.0","description":"Container Storage Interface (CSI) driver, provisioner, snapshotter, and attacher for Ceph RBD","home":"https://github.com/ceph/ceph-csi","keywords":["ceph","rbd","ceph-csi"],"name":"ceph-csi-rbd","sources":["https://github.com/ceph/ceph-csi/tree/csi-v1.0/deploy/rbd/helm"],"version":"0.8.0"}'

unable to decode "../deploy/rbd/kubernetes/v1.14+/helm/values.yaml": Object 'Kind' is missing in '{"attacher":{"affinity":{},"enabled":true,"image":{"pullPolicy":"IfNotPresent","repository":"quay.io/k8scsi/csi-attacher","tag":"v1.2.0"},"name":"attacher","nodeSelector":{},"resources":{},"tolerations":[]},"configMapName":"ceph-csi-config","driverName":"rbd.csi.ceph.com","nodeplugin":{"affinity":{},"name":"nodeplugin","nodeSelector":{},"plugin":{"image":{"pullPolicy":"IfNotPresent","repository":"quay.io/cephcsi/cephcsi","tag":"canary"},"resources":{}},"registrar":{"image":{"pullPolicy":"IfNotPresent","repository":"quay.io/k8scsi/csi-node-driver-registrar","tag":"v1.1.0"},"resources":{}},"tolerations":[]},"pluginDir":"/var/lib/kubelet/plugins","provisioner":{"affinity":{},"image":{"pullPolicy":"IfNotPresent","repository":"quay.io/k8scsi/csi-provisioner","tag":"v1.3.0"},"name":"provisioner","nodeSelector":{},"replicaCount":3,"resources":{},"tolerations":[]},"rbac":{"create":true},"registrationDir":"/var/lib/kubelet/plugins_registry","serviceAccounts":{"attacher":{"create":true,"name":null},"nodeplugin":{"create":true,"name":null},"provisioner":{"create":true,"name":null}},"snapshotter":{"image":{"pullPolicy":"IfNotPresent","repository":"quay.io/k8scsi/csi-snapshotter","tag":"v1.2.0"},"resources":{}},"socketDir":"/var/lib/kubelet/plugins/rbd.csi.ceph.com","socketFile":"csi.sock"}'
```

**Comment by Madhu-1** [comment]
will send a patch to fix it

**Pull request merged**